client/my-sites/email/email-forwarding/email-forwarding-add-new-compact-list.jsx

```diff
@@ -0,0 +1,188 @@
+		withAnalytics(
+			composeAnalytics(
+				recordGoogleEvent(
+					'Domain Management',
+					'Clicked "Add New Email Forward" Button in Email Forwarding',
+					'Domain Name',
+					domainName
+				),
+				recordTracksEvent(
+					'calypso_domain_management_email_forwarding_add_new_email_forward_click',
+					{
+						destination,
+						domain_name: domainName,
+						mailbox,
+					}
+				),
+				this.props.addEmailForward( domainName, mailbox, destination, siteSlug )
+			)
+		);
+
+	addNewEmailForwardsClick = ( event ) => {
+		const { selectedSiteSlug } = this.props;
+
+		event.preventDefault();
+
+		if ( this.state.formSubmitting ) {
+			return;
+		}
+
+		this.setState( { formSubmitting: true } );
+
+		this.state.forwards.map( ( t ) => {
+			const { mailbox, destination } = t;
+
+			this.addNewEmailForwardWithAnalytics(
+				this.props.selectedDomainName,
+				mailbox,
+				destination,
+				selectedSiteSlug
+			);
+		} );
+
+		this.setState( { formSubmitting: false } );
+	};
+
+	onForwardAdd = () => {
+		this.setState( { forwards: [ ...this.state.forwards, { destination: '', mailbox: '' } ] } );
+	};
+
+	addButton( addMoreButton = false ) {
+		const { translate } = this.props;
+		return (
+			<div className="email-forwarding-add-new-compact-list__actions">
+				{ addMoreButton && (
+					<Button
+						className="email-forwarding-add-new-compact-list__add-another-forward-button"
+						onClick={ this.onForwardAdd }
+					>
+						<Gridicon icon="plus" />
+						<span>{ translate( 'Add another forward' ) }</span>
+					</Button>
+				) }
+
+				<Button
+					primary
+					onClick={ this.addNewEmailForwardsClick }
+					disabled={ ! this.validForwards() }
+				>
+					{ translate( 'Add' ) }
+				</Button>
+			</div>
+		);
+	}
+
+	removeButton() {
+		return (
+			<FormButton type="button" isPrimary={ false } onClick={ this.removeClick }>
+				{ this.props.translate( 'Remove' ) }
+			</FormButton>
+		);
+	}
+
+	formFooter() {
+		return <div>{ this.addButton() }</div>;
+	}
+
+	removeHandler = ( index ) => {
+		const array = this.state.forwards;
+		array.splice( index, 1 );
+
+		this.setState( { forwards: array } );
+	};
+
+	updateHandler = ( index, name, value ) => {
+		const array = this.state.forwards;
+		array[ index ][ name ] = value;
+
+		const valid = validateAllFields( array[ index ] );
+		array[ index ].valid = valid.mailbox.length === 0 && valid.destination.length === 0;
+
+		this.setState( { forwards: array } );
+	};
+
+	render() {
+		const { emailForwards, emailForwardingLimit, selectedDomainName } = this.props;
+		const totalFowards = emailForwards.length + this.state.forwards.length;
+		return (
+			<>
+				{ totalFowards > 0 ? (
+					<EmailForwardingLimit
+						emailForwardingCount={ totalFowards }
+						emailForwardingLimit={ emailForwardingLimit }
+					/>
+				) : null }
+				{ this.state.forwards.map( ( fields, index ) => (
+					<Fragment key={ `fragment-${ index }` }>
```

Comment:
Maybe call this something more unique? Same applies for the `form-` and `hr-` keys below.
```suggestion
					<Fragment key={ `email-forward-fragment-${ index }` }>
```